go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -154,6 +154,15 @@ func (vs *vstreamer) Stream() error {
 	}
 	if vs.startPos == "current" {
 		vs.pos = curPos
+		vevents := []*binlogdatapb.VEvent{{
+			Type: binlogdatapb.VEventType_GTID,
+			Gtid: mysql.EncodePosition(vs.pos),
+		}, {
+			Type: binlogdatapb.VEventType_OTHER,
```

Comment:
Why do we need the additional `OTHER` event?

Reply:
Generally, GTIDs should not be recorded by a client until they see a commit. Otherwise, if they fail in the middle of a transaction, they'll lose the rest of the statements on a restart. To allow for this standardization, vstreamer gurarantees that a GTID will always be followed by a COMMIT, OTHER, or DDL. The comment is a few lines below (line 213).

A superior model will be for the GTID to actually be metadata for COMMIT, OTHER, or DDL. But this will become very different from how the binlog reports these.